### PR TITLE
[TASK-1] Implemented File Search

### PR DESCRIPTION
**Pull request by hassanjawwad12**
### Summary
This PR adds the searching implementation of a file in folder using bash. 

### Changes
* Added comprehensive comments.
* Added better echo messages.
* Added string checks for argument testing.
* Added exit after it founds the file name , to stop unnecessary echos.

### Motivation
This enhancement aims to make the code better. 

## How Has This Been Tested?
I tested it by first making the file executable by running chmod +x task.sh 
Then I run it using ./task1.sh profile.jpeg ~/Documents

### CODE REVIEW

File: task1.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#The first thing we will do is check whether the user passed the file and folder name in the args
+#The -z variable tells us if argument is empty string or not 
+if [ -z "$1" ] || [ -z "$2" ]; then
+    echo "Correct Format is following: $0 <file> <folder>"
+    echo "Kindly re-run the script with the correct format"   
+    exit 1
+fi
+
+#we can declare as much arguments as we like here
```
Comment (mzeeshanali) [inline task1.sh]: why we need this comment?

File: task1.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            fi
+        elif [ -d "$item" ]; then
+            #here we are checking the sub directories
+            searching "$item"
+        fi
+    done
+}
+
+searching "$folder"
+
```
Comment (mzeeshanali) [inline task1.sh]: what if the file is not found?

File: task1.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+
+
+
```
Comment (mzeeshanali) [inline task1.sh]: remove these extra lines

File: task1.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+searching() {
+    #local keyword can only be used inside a function, it makes variable name have a visible scope 
+    local folder="$1"
+
+    for item in "$folder"/*; do 
+    #/* is used when we want to search the full folder
+        if [ -f "$item" ]; then
+            if [ "$(basename "$item")" == "$file" ]; then
+                echo "File is found: $item"
+                exit 1 #we ended the loop when file is found
```
Comment (mzeeshanali) [inline task1.sh]: why exit 1? Exit 1 indicates unsuccessful termination

File: task1.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
+<<"COMMENT"
+Write a script to search a file in a given folder.
+~$ ./task1.sh <search_file> <from_folder>
+For example:
+./task1.sh file1 folder1/
+Should output searching every single occurrence of file1 by using depth first search.
+COMMENT
+
+#!/bin/bash
```
Comment (mzeeshanali) [inline task1.sh]: move this to top. It's convention.

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

**Comment by mzeeshanali** [comment]
Please update the PR title.

File: task1.sh (hunk 1 above)
Comment (hassanjawwad12) [reply task1.sh]: I added this comment to tell that we can add more argument by simply doing $3 and so on, if it is unnecessary I will remove it

File: task1.sh (hunk 2 above)
Comment (hassanjawwad12) [reply task1.sh]: I have echoed, that file not found

File: task1.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        elif [ -d "$item" ] && [ "$finding" == "false" ]; then
+            #here we are checking the sub directories
+            searching "$item"
+        fi
+    done
+}
+
+searching "$folder"
+if ! $finding; then
+    echo "Serch is complete but file is not found."
```
Comment (mzeeshanali) [inline task1.sh]:
```suggestion
    echo "Search is complete but file is not found."
```

File: task1.sh [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            fi
+        elif [ -d "$item" ] && [ "$finding" == "false" ]; then
+            #here we are checking the sub directories
+            searching "$item"
+        fi
+    done
+}
+
+searching "$folder"
+if ! $finding; then
```
Comment (mzeeshanali) [inline task1.sh:45]: this will always be false

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

File: task1.sh (hunk 7 above)
Comment (hassanjawwad12) [reply task1.sh:45]: yes lemme fix it

**Pull request merged**